### PR TITLE
Knox replacement

### PR DESCRIPTION
**Pull request by vickywane**
Aws-sdk replacement  for Knox

### CODE REVIEW

**Comment by vickywane** [comment]
Sorry I had to close the former PR.  
I lost my old copy of the project and had to start afresh which lead to some conflicts in the two branches. 

By using the linters I have been able to fix the problem and also running yarn install solves the first issue

**Comment by znarf** [comment]
@vickywane in this project, you should not use `yarn` but `npm install`, and you should commit the modifications to `package-lock.json`.

**Comment by vickywane** [comment]
Sorry about that. 
My personal preference has been Yarn over npm.  
Once I'll push once I make the changes.  

Is yarn responsible for the lint error?

**Comment by znarf** [comment]
@vickywane yes, without the updated `package-lock.json`, we can't install the packages and this is breaking all the tests.

**Comment by vickywane** [comment]
I have updated the required files. 
Can we run the tests again?

**Comment by znarf** [comment]
Unfortunately, the tests are not passing. Have you tried `npx mocha test/images.routes.test.js` ?

**Comment by vickywane** [comment]
@francois I just used the command and went through the file.  
2 failed tests that I don't know how to solve 😔

**Comment by znarf** [comment]
@vickywane you need to make the test work then, if they don't, the feature is likely to be broken.

**Comment by vickywane** [comment]
I refactored the way I used the aws-sdk in my new commit.. The logs shown from the tests do not indicate anything.  I however tested my algorithm in a sandbox and I worked.

**Comment by znarf** [comment]
@vickywane ok, let's try to find more about this error.

**Comment by znarf** [comment]
If you edit the test, you can inspect the error:

```
      .expect(function(res) {
        console.log(res);
      })
      .expect(200)
```

File: server/lib/awsClient.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import config from 'config';
+import aws from 'aws-sdk';
+
+// S3 bucket
+let awsClient;
+
+if (config.aws.s3.key) {
+  awsClient = aws.config.update({
```
Comment (znarf) [inline server/lib/awsClient.js]: Looks like `aws.config.update` is not returning anything, leading to `awsClient` being undefined.

File: server/controllers/images.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import path from 'path';
 import uuidv1 from 'uuid/v1';
+import config from 'config';
 
 import errors from '../lib/errors';
-import knox from '../lib/knox';
+import S3 from '../lib/awsClient';
```
Comment (znarf) [inline server/controllers/images.js]: Inspect `S3`, it should not be `undefined`

File: server/controllers/images.js [hunk 3]
```diff
@@ -35,30 +37,33 @@ export default function uploadImage(req, res, next) {
     );
   }
 
-  if (!knox) {
+  if (!S3) {
```
Comment (znarf) [inline server/controllers/images.js]: This is currently failing here, because `S3` is undefined.

File: server/lib/awsClient.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    accessKeyId: config.aws.s3.key,
+    secretAcessKey: config.aws.s3.secret,
+    region: 'us-west-1',
+  });
+}
+
+const S3 = new awsClient.S3();
```
Comment (znarf) [inline server/lib/awsClient.js]: This is breaking here because `awsClient` is undefined.

**Comment by vickywane** [comment]
Thank you for the head-start.  It would keep working on it

**Comment by vickywane** [comment]
@znarf.  
I turns connection to s3 is initialized using ```js 
new aws.S3({})
```

This change however is not reflected in the docs.  I got it through trial by error.  

I however cannot access the config file hence when I use arbitrary values for the keys everything seems fine.  

I think am enjoying this!

**Comment by znarf** [comment]
Sorry, I can't start reviewing the code itself until the tests are passing. 

- You should address my comments from 3 days ago about `awsClient` being undefined. This is absolutely broken.
- Why did you remove the `multer` package? This is breaking everything.

Looks you're continuously submitting code that is broken and not tested. Before asking for review, the automated tests should pass and you should also test the functionality by yourself.

**Comment by vickywane** [comment]
I didnt make any new commit.  

I only notice that awsClient was being undefined because of a change in the documentation .

The tests errors have changed since then

**Comment by vickywane** [comment]
I installed the multer package in a bid to substitute it with the aws - sdk but that didn't happen so I uninstalled it.  

I totally forgot it being utilised elsewhere

**Comment by vickywane** [comment]
Closing this til all tests pass locally

**Pull request closed without merging**